### PR TITLE
Master announce media chs

### PR DESCRIPTION
**Pull request by nseinlet**

### CODE REVIEW

File: src/util/report.py [hunk 1]
```diff
@@ -60,12 +62,33 @@ def html_escape(text):
...
+
+try:
+    UPG_ANNOUNCE_MEDIA = functools.reduce(
+        int.__or__, (MEDIA[m] for m in split_osenv("UPG_ANNOUNCE_MEDIA", default="discuss"))
+    )
+except (KeyError, ValueError):
+    _logger.error(  # noqa: TRY400
+        "Invalid value for the environment variable `UPG_ANNOUNCE_MEDIA`: %r.\nAuthorized values are a combination of %sor an empty string.",
+        os.getenv("UPG_ANNOUNCE_MEDIA"),
+        ", ".join(sorted(MEDIA, reverse=True)),
```
Comment (nseinlet) [inline src/util/report.py:87]: hq720

**Comment by aj-fuentes** [comment]
From DM: I'd suggest we get rid of the bit-manipulation code and use Python sets instead.
```py
ENV_ANNOUNCE_MEDIA = set(split_osenv("UPG_ANNOUNCE_MEDIA", default="discuss"))
UPG_ANNOUNCE_MEDIA = ENV_ANNOUNCE_MEDIA & {"", "discuss", "logger"}
if ENV_ANNOUNCE_MEDIA - UPG_ANNOUNCE_MEDIA:
    log error ... invalid value ...
...
if "logger" in UPG_ANNOUNCE_MEDIA:
   ...
if not "discuss" in UPG_ANNOUNCE_MEDIA: 
   return
...
```

**Review by aj-fuentes — APPROVED** [review]

**Comment by KangOl** [comment]
upgradeci retry with always only mail

**Review by KangOl — APPROVED** [review]
@robodoo rebase-ff r+

**Pull request closed without merging**